ui/pages/add-collectible/add-collectible.js

```diff
@@ -46,8 +50,68 @@ export default function AddCollectible() {
   );
   const [tokenId, setTokenId] = useState('');
   const [disabled, setDisabled] = useState(true);
+  const [eventTracked, setEventTracked] = useState([
+    {
+      event: 'Token Added',
+      address: '',
+      tracked: false,
+    },
+  ]);
+
   const [collectibleAddFailed, setCollectibleAddFailed] = useState(false);
 
+  const trackEvent = useContext(NewMetaMetricsContext);
+  const collectibles = useSelector(getCollectibles);
+
+  useEffect(() => {
+    if (!tokenId || !address) {
+      return;
+    }
+
+    const newCollectible = collectibles.find(
+      (collectible) =>
+        collectible.address === address &&
+        collectible.tokenId === tokenId.toString(),
+    );
+
+    if (!newCollectible) {
+      return;
+    }
+
+    const alreadyTracked = eventTracked.find(
+      (et) => et.address === address && et.tracked === true,
+    );
+
+    if (alreadyTracked) {
+      return;
+    }
+
+    trackEvent({
+      event: 'Token Added',
+      category: 'Wallet',
+      sensitiveProperties: {
+        token_contract_address: newCollectible.address,        
+        tokenId: newCollectible.tokenId,
+        name: newCollectible.name ? newCollectible.name : null,
```

Comment:
```suggestion
        name: newCollectible.name || null,
```
nit: we could shorten this. Same comment for the `description` prop below

Reply:
fixed [here](https://github.com/MetaMask/metamask-extension/pull/14279/commits/2dff6f9ebd06242a63bd80eec250ec59d245e5e4)